test/common/buffer/buffer_test.cc

```diff
@@ -1075,6 +1076,23 @@ TEST(BufferHelperTest, WriteBEI64) {
     EXPECT_EQ("\x7F\xFF\xFF\xFF\xFF\xFF\xFF\xFF", buffer.toString());
   }
 }
+TEST(BufferHelperTest, AddFragments) {
+  {
+    // Add some string fragments.
+    Buffer::OwnedImpl buffer;
+    buffer.addFragments("aaaaa", absl::string_view("bbbbb"));
+    EXPECT_EQ("aaaaabbbbb", buffer.toString());
+  }
+
+  {
+    // Add string fragments cyclically.
+    Buffer::OwnedImpl buffer;
+    for (size_t i = 0; i < 1024; i++) {
+      buffer.addFragments("aaaaa", "bbbbb", "ccccc", "ddddd");
+    }
+    EXPECT_EQ(buffer.length(), 20 * 1024);
+  }
+}
```

Comment:
Use expectSlices to verify the structure of the slices inside the buffer.  Since you're writing 20 bytes at a time, you should see five 4080 byte slices and one 80 byte slice.

Reply:
get it.